pgcli/main.py

```diff
@@ -79,9 +78,28 @@ def __init__(self, force_passwd_prompt=False, never_passwd_prompt=False,
         # Initialize completer
         smart_completion = c['main'].as_bool('smart_completion')
         completer = PGCompleter(smart_completion)
-        completer.extend_special_commands(CASE_SENSITIVE_COMMANDS.keys())
-        completer.extend_special_commands(NON_CASE_SENSITIVE_COMMANDS.keys())
+        completer.extend_special_commands(COMMANDS.keys())
         self.completer = completer
+        self.register_special_commands()
+
+    def register_special_commands(self):
```

Comment:
I can't replace it with a decorator because when you decorate a method that is in a class, the method is still unbound (it is not attached to an instance). So when the execute() methods looks up the special command in the COMMANDS dict, it'll get back an unbound method which can't be used. We need the actual instance so we can call change_db on the instance of the PGCli class. I'm not sure if I made that point clear. If it's not clear please let me know and I'll try to rephrase it.

I like the idea of passing in the aliases. I'm going to make the aliases always hidden. Because if you have two shortcuts doing the same thing then you don't need them both listed in the help command. Even if we want to list them both we can instead use a second decorator like here: https://github.com/dbcli/pgcli/pull/255/files#diff-f353ad4995ec86e0878d2a80b589c6daR288. Because there is only one 'syntax' and 'description' field and I can't display those for aliases.